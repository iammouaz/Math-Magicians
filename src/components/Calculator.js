/* eslint-disable react/destructuring-assignment */
const Calc = (int) => (
  <div className={int.class}>
    {int.number}
  </div>
);

const App = () => (
  <div className="Calculator">
    <input className="input" value="0" />
    <Calc class="num" number="AC" />
    <Calc class="num" number="+/-" />
    <Calc class="num" number="%" />
    <Calc class="num math" number="รท" />
    <Calc class="num" number="7" />
    <Calc class="num" number="8" />
    <Calc class="num" number="9" />
    <Calc class="num math" number="ร" />
    <Calc class="num" number="4" />
    <Calc class="num" number="5" />
    <Calc class="num" number="6" />
    <Calc class="num math" number="-" />
    <Calc class="num" number="1" />
    <Calc class="num" number="2" />
    <Calc class="num" number="3" />
    <Calc class="num math" number="+" />
    <Calc class="num zero" number="0" />
    <Calc class="num" number="." />
    <Calc class="num math" number="=" />
  </div>
);

export default App;
