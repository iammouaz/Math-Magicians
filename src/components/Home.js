import React from 'react';

const Home = () => (
  <div className="homepage">
    <h2 className="welcome">Welcome to Home Page</h2>
    <p className="this">This is a website for all fans of mathematics. It is a Single Page App (SPA) that allows users to make simple calculations and to read a random math-related quote. Built using React and Bootstrap. </p>
  </div>
);

export default Home;
