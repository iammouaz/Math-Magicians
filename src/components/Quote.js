import React from 'react';

const Quote = () => (
  <div>
    <h3 className="quote">
      Mathematics is not about numbers, equations,
      computations, or algorithms it is about understandings. -William Paul Thurston
    </h3>
  </div>
);

export default Quote;
