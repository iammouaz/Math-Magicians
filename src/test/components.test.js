import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

// components
import Home from '../components/Home';
import Calculator from '../components/Calculator';
import Quote from '../components/Quote';

describe('Components Matches Test', () => {
  it('Home Components Matches', () => {
    const home = renderer
      .create(<Home />)
      .toJSON();
    expect(home).toMatchSnapshot();
  });

  it('Calculator Components Matches', () => {
    const calculator = renderer
      .create(<Calculator />)
      .toJSON();
    expect(calculator).toMatchSnapshot();
  });

  it('quote Components Matches', () => {
    const quote = renderer
      .create(<Quote />)
      .toJSON();
    expect(quote).toMatchSnapshot();
  });
});

describe('User interaction simulation test', () => {
  it('Home route test', () => {
    render(<Home />);
  });
  it('Quote route test', () => {
    render(<Quote />);
  });
  it('Calculator route test', () => {
    render(<Calculator />);
  });
});
