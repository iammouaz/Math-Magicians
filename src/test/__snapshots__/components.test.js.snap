// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Matches Test Calculator Components Matches 1`] = `
<section
  className="Calc"
>
  <div
    className="Lets"
  >
    <h2
      className="do"
    >
      Lets Do Some Math!
    </h2>
  </div>
  <div
    className="Calculator"
  >
    <span
      className="input"
    >
      <span
        className="number"
      >
        0
      </span>
    </span>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="%"
    >
      %
    </button>
    <button
      className="num math"
      onClick={[Function]}
      type="button"
      value="รท"
    >
      รท
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="7"
    >
      7
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="8"
    >
      8
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="9"
    >
      9
    </button>
    <button
      className="num math"
      onClick={[Function]}
      type="button"
      value="x"
    >
      ร
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="4"
    >
      4
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="5"
    >
      5
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="6"
    >
      6
    </button>
    <button
      className="num math"
      onClick={[Function]}
      type="button"
      value="-"
    >
      -
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="1"
    >
      1
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="2"
    >
      2
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="3"
    >
      3
    </button>
    <button
      className="num math"
      onClick={[Function]}
      type="button"
      value="+"
    >
      +
    </button>
    <button
      className="num zero"
      onClick={[Function]}
      type="button"
      value="0"
    >
      0
    </button>
    <button
      className="num"
      onClick={[Function]}
      type="button"
      value="."
    >
      .
    </button>
    <button
      className="num math"
      onClick={[Function]}
      type="button"
      value="="
    >
      =
    </button>
  </div>
</section>
`;

exports[`Components Matches Test Home Components Matches 1`] = `
<div
  className="homepage"
>
  <h2
    className="welcome"
  >
    Welcome to Home Page
  </h2>
  <p
    className="this"
  >
    This is a website for all fans of mathematics. It is a Single Page App (SPA) that allows users to make simple calculations and to read a random math-related quote. Built using React and Bootstrap. 
  </p>
</div>
`;

exports[`Components Matches Test quote Components Matches 1`] = `
<div>
  <h3
    className="quote"
  >
    Mathematics is not about numbers, equations, computations, or algorithms it is about understandings. -William Paul Thurston
  </h3>
</div>
`;
